agent/internal/container/container.go

```diff
@@ -218,9 +218,8 @@ func (c *Container) run(parent context.Context) (err error) {
 			if err != nil {
 				c.log.Trace("ensuring cleanup of container (canceled prior to the monitoring loop)")
 				if remove {
-					// TODO this context could be canceled between CreateContainer and this.
-					if rErr := c.docker.RemoveContainer(ctx, dockerID, true); rErr != nil {
-						c.log.WithError(rErr).Debug("couldn't cleanup container")
+					if rErr := c.docker.RemoveContainer(parent, dockerID, true); rErr != nil {
+						c.log.WithError(rErr).Debug("attempted to use parent context, couldn't cleanup container")
```

Comment:
remove the attempted to use parent context part of log

Reply:
Sounds good, especially since now there's no need to make the distinction and it doesn't add anything for the end user besides more questions haha